full_bubble_model.py

```diff
@@ -15,20 +19,104 @@
 try:
     import parameters as par   # numeric constants and coefficents
     importlib.reload(par)   # reload changes you made
-    print(f'model: {par.model}')
 except:
-    print('Error, \'parameters.py\' not found')
+    print(print(colored('Error, \'parameters.py\' not found','red')))
+try:
+    import excitation
+    importlib.reload(excitation)
+except:
+    try:
+        import Bubble_dynamics_simulation.excitation as excitation
+        importlib.reload(excitation)
+    except:
+        print(colored(f'Error, \'excitation.py\' not found', 'red'))    
+
+"""________________________________Settings________________________________"""
+
+enable_heat_transfer = True
+enable_evaporation = False
+enable_reactions = True
+enable_dissipated_energy = False
+target_specie = 'NH3' # Specie to calculate energy effiqiency
+excitation_type = 'no_excitation' # function to calculate pressure excitation
+
+"""________________________________General________________________________"""
 
-# dot.notation access to dictionary attributes
-# instead of dictionary['key'] you can use dictionary.key
 class dotdict(dict):
+    """Dot notation access to dictionary attributes. 
+    Instead of dictionary['key'] you can use dictionary.key"""
     __getattr__ = dict.get
     __setattr__ = dict.__setitem__
     __delattr__ = dict.__delitem__
-    
+
+def copy(input):
```

Comment:
This function deep copies anything. This will remove referencing. What is deep copy?
If you make any complex data structure, like a list: a=[1, 2, 3]
then it will be handled as a reference (pointer), like for example when it is passed into a function as argument.
Therefore if you change it in the function, the outside list changes as well. E.g.: b=a; b.append(4)
Now a is also appended: print(a) --> [1, 2, 3, 4]
To avoid this behaviour, use b=de.copy(a)